source/model/matchStatus.mc

```diff
@@ -57,6 +61,10 @@ class MatchStatus {
         self.p1ScoreIdx++;
     }
 
+    function getP1Idx() {
```

Comment:
i think this change happened because we want to display the G-G score when in golden point right?  i feel we should revert it. instead of doing this, can we infer we are in the G-G state by looking at the score and at the number of deuces?

Reply:
> i think this is somehow new. the p1Idx and p2Idx were not exposed on purpose. it's an internal data structure that should not be used by anyone else other than the matchStatus class. callers should look into the score only

How come not here? That's the file where I found all the functions, it seemed more practical to draw from here since you have an index with all the scores.
So if I understand you correctly, you would rather create a second function that does the calculations and just returns the value? How were you planning to proceed?

Reply:
> i think this change happened because we want to display the G-G score when in golden point right? i feel we should revert it. instead of doing this, can we infer we are in the G-G state by looking at the score and at the number of deuces?

Trying this feature, I realized that a visual indication is better, because it happens that the point lasts a very long time, so long that you don't remember accurately the score, which is then one of the purposes of the application. I think it's happened to you playing as well, points of a long duration that at the conclusion you don't know the score. So the ploy, you run out of deuces available, the application points to G-G which reminds you that you are at the Golden Point and the point is decisive. The same thing happens with A if you think about it.
I would take advantage of the dedicated array of scores and "play" by moving the index time by time.
How would you do that?